libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -446,7 +463,15 @@ void AP_Camera::update_trigger()
         DataFlash_Class *df = DataFlash_Class::instance();
         if (df != nullptr) {
             if (df->should_log(log_camera_bit)) {
-                df->Log_Write_Camera(ahrs, current_loc);
+                uint64_t timestamp = 0;
+                if (_feedback_sem && _feedback_sem->take_nonblocking()) {
+                    timestamp = _feedback_timestamp_us;
+                    _feedback_sem->give();
+                }
+                if (timestamp == 0) {
```

Comment:
This can be avoided, given that sending 0 means dataflash takes care of using the time.